org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -895,6 +898,10 @@ protected CompletableFuture<LanguageServer> getInitializedServer() {
 		return CompletableFuture.completedFuture(context.languageServer);
 	}
 
+	public ServerInfo getInitializedServerInfo() {
+		return initializedServerInfo;
```

Comment:
Should it 1st join on initializeFuture being complete?
Or maybe return a `CompletionFuture<ServerInfo>` with `return initializeFuture.thenApply(_ -> this.initializeServerInfo)` ?